src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/SimpleStrandSwitchVariantDetector.java

```diff
@@ -0,0 +1,301 @@
+     */
+    private static Tuple2<ChimericAlignment, byte[]> convertAlignmentIntervalToChimericAlignment
+    (final AlignedContig longReadWith2AIMappedToSameChrAndStrandSwitch) {
+
+        final AlignmentInterval intervalOne = longReadWith2AIMappedToSameChrAndStrandSwitch.alignmentIntervals.get(0),
+                intervalTwo = longReadWith2AIMappedToSameChrAndStrandSwitch.alignmentIntervals.get(1);
+
+        if (splitPairStrongEnoughEvidenceForCA(intervalOne, intervalTwo, MORE_RELAXED_ALIGNMENT_MIN_MQ,  MORE_RELAXED_ALIGNMENT_MIN_LENGTH)) {
+            return new Tuple2<>(new ChimericAlignment(intervalOne, intervalTwo, EMPTY_INSERTION_MAPPINGS,
+                    longReadWith2AIMappedToSameChrAndStrandSwitch.contigName), longReadWith2AIMappedToSameChrAndStrandSwitch.contigSequence);
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Roughly similar to {@link ChimericAlignment#nextAlignmentMayBeNovelInsertion(AlignmentInterval, AlignmentInterval, Integer)}:
+     *  1) either alignment may have very low mapping quality (a more relaxed mapping quality threshold);
+     *  2) either alignment may consume only a "short" part of the contig, or if assuming that the alignment consumes
+     *     roughly the same amount of ref bases and read bases, has isAlignment that is too short
+     */
+    private static boolean splitPairStrongEnoughEvidenceForCA(final AlignmentInterval intervalOne,
+                                                              final AlignmentInterval intervalTwo,
+                                                              final int mapQThresholdInclusive,
+                                                              final int alignmentLengthThresholdInclusive) {
+
+        if (intervalOne.mapQual < mapQThresholdInclusive || intervalTwo.mapQual < mapQThresholdInclusive)
+            return false;
+
+        final int overlap = AlignmentInterval.overlapOnContig(intervalOne, intervalTwo);
+
+        final int x = intervalOne.endInAssembledContig - intervalOne.startInAssembledContig + 1,
+                y = intervalTwo.endInAssembledContig - intervalTwo.startInAssembledContig + 1;
+
+        return Math.min(x - overlap, y - overlap) >= alignmentLengthThresholdInclusive;
+    }
+
+    // =================================================================================================================
+
+    private JavaRDD<VariantContext> dealWithSimpleStrandSwitchBkpts(final JavaRDD<AlignedContig> longReads,
+                                                                    final Broadcast<ReferenceMultiSource> broadcastReference,
+                                                                    final Logger toolLogger) {
+
+        final JavaPairRDD<ChimericAlignment, byte[]> simpleStrandSwitchBkpts =
+                longReads
+                        .mapToPair(SimpleStrandSwitchVariantDetector::convertAlignmentIntervalToChimericAlignment)
```

Comment:
To me a more straightforward flow would be to `filter` them on `splitPairStrongEnoughEvidenceForCA` and then convert the remaining ones to `ChimericAlignment`s, rather than filtering on `nonNull`.